src/components/common/LoadingPage.js

```diff
@@ -10,11 +10,38 @@ import "styles/Loading.scss";
 		textPadding: string representing padding to the left of the text, i.e. distance from the img (give px units)
 */
 
-const Loading = props => (
-  <div className="Loading">
-    <div className="Loading-title">Loading</div>
-    <RingLoader color={"#171124"} size={250} loading={true} />
-  </div>
-);
+class Loading extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showText: false,
+      timer: 0,
+    };
+  }
+  componentDidMount = () => {
+    this.setState.timer = setTimeout(() => {
+      this.setState({ showText: true });
+    }, 2000);
+  };
+
+  componentWillUnmount = () => {
+    clearTimeout(this.setState.timer);
+  };
+
+  render = () => {
+    return (
+      <div className="Loading">
+        <div className="Loading-title">Loading</div>
+        <RingLoader color={"#171124"} size={250} loading={true} />
+        {this.state.showText && (
+          <p className="Loading-page-text" style={{ color: "white" }}>
+            Looks like loading is taking a bit long! If it takes too long, submit an issue on
+            <a href="https://github.com/uclaacm/TeachLAFrontend/issues"> Github</a>.
```

Comment:
This link doesn't contrast too well with the purple background. I'd probably create a class called `.Loading-link-text` (or something like that), and use

```
.Loading-link-text {
  color: $off-white;
  text-decoration: underline;
  ...
}
```

Which would make it contrast better with the background but still make it obvious that it's a link. You might also want to take a look at the `:hover` attribute, which will also need some styling.